File: Part 3/SE3AC11/src/cspmodel/CarPark.java
package cspmodel;

import org.jcsp.lang.CSProcess;
import org.jcsp.lang.Channel;
import org.jcsp.lang.One2OneChannelInt;
import org.jcsp.lang.Parallel;

public class CarPark {

	public static void main(String args[]){

		One2OneChannelInt arrive = Channel.one2oneInt(); 
		One2OneChannelInt depart = Channel.one2oneInt();

		Parallel carpark =  new Parallel(new CSProcess[] {
				new Arrives(arrive),
				new Control(arrive, depart),
				new Departs(depart)
		});	
		carpark.run();

	}
}

File: Part 3/SE3AC11/CarPark/carpark.py
import sys

from pycsp import *
import carpark_arrival
import carpark_departs
import carpark_control

@process
def run():
	arrives = Channel()
	departs = Channel()

	Parallel(
		carpark_arrival.run(+arrives),
		carpark_control.run(-arrives, -departs),
		carpark_departs.run(+departs)
	)